### ISSUE TITLE
Make `// eslint-disable-next-line` directives hoverable with a link to the rule

### ISSUE THREAD

**Issue by thernstig** [microsoft/vscode-eslint#2187]
Example:

```ts
  const scriptDirectory = import.meta.dirname;
  const outputDirectoryPath = path.resolve(scriptDirectory, 'output');

  await fs.mkdir(outputDirectoryPath, { recursive: true });

  const timestamp = new Date().toISOString().replaceAll(/[.:]/g, '-');
  const outputPath = path.resolve(outputDirectoryPath, `marketstack-${timestamp}.json`);

  // eslint-disable-next-line security/detect-non-literal-fs-filename -- This is flagged on new Date() which is harmless and necessary for timestamping the output file.
  await fs.writeFile(outputPath, JSON.stringify(completeResponse, undefined, 2));
  console.log(`✓ Full response saved to: ${outputPath}`);
```

It would be nice to be able to hover over the line with `eslint-disable-next-line security/detect-non-literal-fs-filename` brings me to the rule page. The same way the underline that happens if I remove the disable line, which I can hover to get a popup to click the link to the rule.

**Comment by dbaeumer** [comment]
From an implementation perspective the eslint server would need to provider a hover provider that returns results when hovering of a // eslint-disable text.
